openshift/catalogd/generate-manifests.sh

```diff
@@ -0,0 +1,132 @@
+  $YQ -i 'select(.kind == "Deployment").spec.template.metadata.annotations += {"target.workload.openshift.io/management": "{\"effect\": \"PreferredDuringScheduling\"}"}' "$TMP_KUSTOMIZE_OUTPUT"
+  $YQ -i 'select(.kind == "Deployment").spec.template.metadata.annotations += {"openshift.io/required-scc": "privileged"}' "$TMP_KUSTOMIZE_OUTPUT"
+  $YQ -i 'select(.kind == "Deployment").spec.template.spec += {"priorityClassName": "system-cluster-critical"}' "$TMP_KUSTOMIZE_OUTPUT"
+  $YQ -i 'select(.kind == "Namespace").metadata.annotations += {"workload.openshift.io/allowed": "management"}' "$TMP_KUSTOMIZE_OUTPUT"
+done
+
+# Loop through any flag updates that need to be made to the manager container
+for flag_name in "${!FLAG_MAPPINGS[@]}"; do
+  flagval="${FLAG_MAPPINGS[$flag_name]}"
+
+  # First, update the flag if it exists
+  $YQ -i "(select(.kind == \"Deployment\") | .spec.template.spec.containers[] | select(.name == \"manager\") | .args[] | select(. | contains(\"--$flag_name=\")) | .) = \"--$flag_name=$flagval\"" "$TMP_KUSTOMIZE_OUTPUT"
+
+  # Then, append the flag if it doesn't exist
+  $YQ -i "(select(.kind == \"Deployment\") | .spec.template.spec.containers[] | select(.name == \"manager\") | .args) |= (select(.[] | contains(\"--$flag_name=\")) | .) // . + [\"--$flag_name=$flagval\"]" "$TMP_KUSTOMIZE_OUTPUT"
+done
+
+# Use yq to split the single yaml file into 1 per document.
+# Naming convention: $index-$kind-$namespace-$name. If $namespace is empty, just use the empty string.
+(
+  cd "$TMP_MANIFEST_DIR"
+
+  # shellcheck disable=SC2016
+  ${YQ} -s '$index +"-"+ (.kind|downcase) +"-"+ (.metadata.namespace // "") +"-"+ .metadata.name' temp.yaml
+)
+
+# Delete the single yaml file
+rm "$TMP_KUSTOMIZE_OUTPUT"
+
+# Delete and recreate the actual manifests directory
+MANIFEST_DIR="${REPO_ROOT}/openshift/catalogd/manifests"
+rm -rf "${MANIFEST_DIR}"
+mkdir -p "${MANIFEST_DIR}"
+
+# Copy everything we just generated and split into the actual manifests directory
+cp "$TMP_MANIFEST_DIR"/* "$MANIFEST_DIR"/
+
+# Update file names to be in the format nn-$kind-$namespace-$name
+(
+  cd "$MANIFEST_DIR"
+
+  for f in *; do
+    # Get the numeric prefix from the filename
+    index=$(echo "$f" | cut -d '-' -f 1)
+    # Keep track of the full file name without the leading number and dash
+    name_without_index=${f#$index-}
+    # Fix the double dash in cluster-scoped names
+    name_without_index=${name_without_index//--/-}
+    # Reformat the name so the leading number is always padded to 2 digits
+    new_name=$(printf "%02d" "$index")-$name_without_index
+    # Some file names (namely CRDs) don't end in .yml - make them
+    if ! [[ "$new_name" =~ yml$ ]]; then
+      new_name="${new_name}".yml
+    fi
+    if [[ "$f" != "$new_name" ]]; then
+      # Rename
+      mv "$f" "${new_name}"
+    fi
+  done
+)
```

Comment:
I am assuming that it is a copy from https://github.com/openshift/operator-framework-catalogd/blob/main/openshift/generate-manifests.sh. So not going through the details.

Reply:
Yes :-) 

 it was copied from there and adjusted accordingly. The same changes we applied to the operator-controller manifests were made here as well, since the OpenShift manifests are now located under openshift/catalogd, and the catalogd project manifests are in the ROOT/catalogd directory instead of ROOT.

mainly the same changes done for operator-controller here: https://github.com/openshift/operator-framework-operator-controller/pull/232/commits/131d4c4057511ed000dbf65a35b034feb375d0b2 are applied for Openshift manifest of catalogd.